gems/aws-sdk-core/lib/aws-sdk-core/plugins/protocols/rest_json.rb

```diff
@@ -5,7 +5,31 @@ module Plugins
     module Protocols
       class RestJson < Seahorse::Client::Plugin
 
+        class ContentTypeHandler < Seahorse::Client::Handler
+          def call(context)
```

Comment:
Rather than checking for `modeled_body?` here (and having to duplicate the search essentially) - could we ensure that this handler runs after the Rest::Handler, and then just check if a body is set - that we we reduce the number of places that need to understand/inspect the model.

Reply:
Yeah, that can be done.